frontend/src/components/Test.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from "react";
+import {
+  fetchPlaylists,
+  getAccessToken,
+  getRefreshToken,
+} from "../services/spotifyApi";
+import type { TokenInfo, PlayList } from "../types/playlists";
+export function Test() {
+  const [playlists, setPlaylists] = useState<PlayList[]>([]);
+  const [tokenAge, setTokenAge] = useState<boolean>(false);
+
+  const persistAccessToken = async (code: string) => {
+    try {
+      const getAccessTokenResponse: TokenInfo = {
+        ...(await getAccessToken(code)),
+        date: Date.now(),
+      };
+
+      if (!getAccessTokenResponse || !getAccessTokenResponse.access_token) {
+        throw new Error("Invalid Access Token Response");
+      }
+
+      localStorage.setItem("tokenInfo", JSON.stringify(getAccessTokenResponse));
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const switchTokens = async (refreshToken: string) => {
+    try {
+      const getAccessTokenResponse: TokenInfo = {
+        ...(await getRefreshToken(refreshToken)),
+        date: Date.now(),
+        refresh_token: refreshToken,
+      };
+
+      if (!getAccessTokenResponse || !getAccessTokenResponse.access_token) {
+        throw new Error("Invalid Access Token Response");
+      }
+
+      localStorage.setItem("tokenInfo", JSON.stringify(getAccessTokenResponse));
+      console.log("Tokens have been exchanged for a fresh one");
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const getPlaylists = async (accessToken: string, limit: number) => {
+    try {
+      const playlistResponse = await fetchPlaylists(accessToken, 10);
+      return playlistResponse.items;
+    } catch (error) {
+      return [];
+    }
+  };
+
+  const tokenIsExpired = (tokenDate: number): boolean => {
+    let currentDate = Date.now();
+    let tokenAge = (currentDate - tokenDate) / 1000;
+    return tokenAge >= 3600 ? true : false;
+  };
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
+    const code = params.get("code") as string;
+
+    const tokenInfoJson = JSON.parse(
+      localStorage.getItem("tokenInfo")!
+    ) as TokenInfo;
+
+    if (
+      (tokenInfoJson === null || tokenInfoJson === undefined) &&
+      tokenAge === false
+    ) {
+      persistAccessToken(code);
+    }
+
+    if (tokenInfoJson) {
+      const tokenStatus = () => {
+        setTokenAge(tokenIsExpired(tokenInfoJson.date));
+      };
+      tokenStatus();
+      if (tokenAge === true) {
+        switchTokens(tokenInfoJson.refresh_token);
+      }
+    }
+
+    if (tokenInfoJson && tokenAge === false) {
+      const test = async () => {
+        setPlaylists(await getPlaylists(tokenInfoJson.access_token, 10));
+      };
+      test();
+    }
+
+    if (!code || tokenInfoJson !== null) {
+      return;
+    }
+  }, [tokenAge]);
+  return (
+    <div>
+      {tokenAge && tokenAge}
+      {playlists &&
+        playlists.map((playlist, index) => (
```

Comment:
```suggestion
        playlists?.map((playlist, index) => (
```

safe navigation here should prevent an undefined error if thats what we're guarding against